File: SDKs/googleapis.com/run/v1/java/lib/src/main/java/openapisdk/models/shared/DomainMapping.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * DomainMapping
 * Resource to hold the state and status of a user's domain mapping. NOTE: This resource is currently in Beta.
**/
public class DomainMapping {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("apiVersion")
    public String apiVersion;
    public DomainMapping withApiVersion(String apiVersion) {
        this.apiVersion = apiVersion;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("kind")
    public String kind;
    public DomainMapping withKind(String kind) {
        this.kind = kind;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("metadata")
    public ObjectMeta metadata;
    public DomainMapping withMetadata(ObjectMeta metadata) {
        this.metadata = metadata;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("spec")
    public DomainMappingSpec spec;
    public DomainMapping withSpec(DomainMappingSpec spec) {
        this.spec = spec;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("status")
    public DomainMappingStatus status;
    public DomainMapping withStatus(DomainMappingStatus status) {
        this.status = status;
        return this;
    }
}

File: SDKs/googleapis.com/run/v1/python/src/sdk/models/shared/domainmapping.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import objectmeta as shared_objectmeta
from ..shared import domainmappingspec as shared_domainmappingspec
from ..shared import domainmappingstatus as shared_domainmappingstatus


@dataclass_json
@dataclasses.dataclass
class DomainMapping:
    r"""DomainMapping
    Resource to hold the state and status of a user's domain mapping. NOTE: This resource is currently in Beta.
    """
    
    api_version: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('apiVersion') }})
    kind: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('kind') }})
    metadata: Optional[shared_objectmeta.ObjectMeta] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }})
    spec: Optional[shared_domainmappingspec.DomainMappingSpec] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('spec') }})
    status: Optional[shared_domainmappingstatus.DomainMappingStatus] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    
